pycytominer/aggregate.py

```diff
@@ -49,6 +53,7 @@ def aggregate(
 
     # Subset dataframe to only specified variables if provided
     strata_df = population_df.loc[:, strata]
+    count_object_df = population_df.loc[:, strata + [object_feature]]
```

Comment:
Can you move this line to only execute if `compute_object_count=True`? I think it shouldn't execute unless we use the `count_object_df` variable.

Reply:
I wasn't able to because `population_df` does not have all the metadata fields at this point (i.e. no `object_feature`). So I have done something like [this](https://github.com/cytomining/pycytominer/pull/129/commits/45684d4a9b1ac4ea61bbd5082cffcdde0d46175e) but this looks clunky as well.

Reply:
oh i see the issue. Yeah, I also agree that the current solution is clunky. We shouldn't create a copy of input data unless we absolutely need to, and especially not by default. Imagine running this function on really large data - watch out!

I think that we can avoid this issue by instead using two `if compute_object_count` statements.

One after defining `strata_df`:

```python
strata_df = population_df.loc[:, strata]

# Only extract single object column in preparation for count
if compute_object_count:
    count_object_df = population_df_copy.loc[:, strata + [object_feature]]
    count_object_df = (
        count_object_df.groupby(strata)[object_feature]
        .count()
        .reset_index()
        .rename(columns={f"{object_feature}": f"Metadata_Object_Count"})
    )

if features == "infer":
    features = infer_cp_features(population_df)
    population_df = population_df.loc[:, features]
else:
    population_df = population_df.loc[:, features]
```

Then, after performing the actual aggregation:

```python
if compute_object_count:
    population_df = count_object_df.merge(population_df, on=strata, how="right")
```

How does this sound? I'd prefer a solution avoiding duplicate data while maintaining readability.

Reply:
This makes sense! Implemented your suggestion -https://github.com/cytomining/pycytominer/pull/129/commits/3dcbe344595981cf8fd45727681c4b8fff617f88